### PR TITLE
Encode markerName with UCS2

### PR DESCRIPTION
**Pull request by mycheckgithub**

### CODE REVIEW

**Comment by RemiArnaud** [comment]
let's make this an exporter option to encode ALL xsd:name to valid character set.
User has choice to type in valid names, and therefore have 1:1 correspondence and human readable.
Off by default 'should invalid names be encoded'

File: COLLADABaseUtils/src/COLLADABUStringUtils.cpp [hunk 1]
```diff
@@ -131,6 +204,65 @@ namespace COLLADABU
...
+		const char DEC2HEX[16 + 1] = "0123456789ABCDEF";
+		const unsigned char * pSrc = (const unsigned char *)sSrc.c_str();
+		const int SRC_LEN = (const int)sSrc.length();
+		unsigned char * const pStart = new unsigned char[SRC_LEN * 7];
+		unsigned char * pEnd = pStart;
+		const unsigned char * const SRC_END = pSrc + SRC_LEN;
+
+		for (; pSrc < SRC_END; ++pSrc)
+		{
+			*pEnd++ = '_';
```
Comment (RemiArnaud) [inline COLLADABaseUtils/src/COLLADABUStringUtils.cpp]: where did you find this ?
this does not look like XML escape sequence ?

&#DDD;  where DDD is the decimal number of the character
or 
&#xHH; where HH is the hexadecimal

File: COLLADABaseUtils/include/COLLADABUStringUtils.h [hunk 2]
```diff
@@ -108,6 +127,8 @@ namespace COLLADABU
 		static char toUpperASCIIChar( char c );
 
         static String uriEncode ( const String & sSrc );
+		static String ucs2Encode(const String & sSrc);
```
Comment (RemiArnaud) [inline COLLADABaseUtils/include/COLLADABUStringUtils.h]: We're not encoding URIs here, we're encoding XML names.
That's not the same encoding/decoding scheme at all.

**Comment by mycheckgithub** [comment]
+ Added explanation about name encoding in README
+ Added ToolTip in UI option"Encode Names"
+ Fix encoding colon

File: COLLADABaseUtils/src/COLLADABUStringUtils.cpp (hunk 1 above)
Comment (mbarnes-sb) [inline COLLADABaseUtils/src/COLLADABUStringUtils.cpp]: okay understood. makes sense. :)

File: COLLADABaseUtils/src/COLLADABUStringUtils.cpp [hunk 3]
```diff
@@ -25,39 +25,112 @@ namespace COLLADABU
...
+			const wchar_t& character4 = ncName[i + 3];
+			const wchar_t& character5 = ncName[i + 4];
+			const wchar_t& character6 = ncName[i + 5];
+			const wchar_t& character7 = ncName[i + 6];
+
+			if ((character == '_') &&
+				(character2 == 'x') &&
+				(isAsciiAlphaChar(character3) || isDigit(character3)) &&
+				(isAsciiAlphaChar(character4) || isDigit(character4)) &&
+				(isAsciiAlphaChar(character5) || isDigit(character5)) &&
```
Comment (russoheney) [inline COLLADABaseUtils/src/COLLADABUStringUtils.cpp]: (isAsciiAlphaChar(character6) || isDigit(character6)) ?

Comment (mycheckgithub) [inline COLLADABaseUtils/src/COLLADABUStringUtils.cpp]: Need to fix

**Pull request merged**